src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionInfo.cs

```diff
@@ -99,6 +99,15 @@ public ICollection<string> AllConnectionTypes
             }
         }
 
+        public bool HasConnectionType(string connectionType)
+        {
+            if (connectionType == null)
+            {
+                connectionType = ConnectionType.Default;
+            }
+            return ConnectionTypeToConnectionMap.ContainsKey(connectionType);

```

Comment:
So one of the really nice features of ConcurrentDictionary is that it supports having a function called to update an item in the dictionary. So if you're doing something like you say (check for item with given connection type, if it exists disconnect it and add a new one) you can do it pretty neatly with

```
string connType = connectionType ?? ConnectionType.Default;
ConnectionInfo newCi = new ConnectionInfo(); //whatever you do to create one
ConnectionTypeToConnectionMap.AddOrUpdate(connType, newCi, (key, oldCi) => {
  oldCi.Disconnect(); // or whatever else you need to do to it
  return newCi;
});
```